library/src/main/java/com/hannesdorfmann/adapterdelegates3/ListDelegationAdapter.java

```diff
@@ -35,19 +37,109 @@
  * }
  * </pre>
  *
- * @param <T> The type of the items. Must be something that extends from List like List<Foo>
+ * @param <T> The type of the list items of this adapter
  * @author Hannes Dorfmann
  */
-public class ListDelegationAdapter<T extends List<?>> extends AbsDelegationAdapter<T> {
+public class ListDelegationAdapter<T> extends AbsDelegationAdapter<List<T>> {
 
   public ListDelegationAdapter() {
   }
 
-  public ListDelegationAdapter(@NonNull AdapterDelegatesManager<T> delegatesManager) {
+  public ListDelegationAdapter(@NonNull AdapterDelegatesManager<List<T>> delegatesManager) {
     super(delegatesManager);
   }
 
   @Override public int getItemCount() {
     return items == null ? 0 : items.size();
   }
+
+
+  /**
+   * Called to determine if data source of this adapter is empty.
+   * <p>
+   * Proxy method for {@link List#isEmpty}
+   *
+   * @return true, if dataset is empty, otherwise false
+   */
+  public boolean isEmpty() {
+    return items.isEmpty();
+  }
+
+
+  /**
+   * Returns the element at the specified position in this adapter's List
+   * <p>
+   * Proxy method for {@link List#get(int)}
+   */
+  public T getItem(int pos) {
+    return items.get(pos);
+  }
+
+
+  /**
+   * Appends the specified item to the List of this adapter
+   * <p>
+   * Proxy method for {@link List#add(Object)}
+   */
+  public void addItem(T item) {
+    items.add(item);
+  }
+
+  /**
+   * Inserts the specified item at the specified position in the List of this adapter
+   * <p>
+   * Proxy method for {@link List#add(int, Object)}
+   */
+  public void addItem(int pos, T item) {
```

Comment:
Im not a fan of mutable data. Also, this  might breaks kotlin compatibility since kotlin `List` is immutable. I'm sorry, but I cant merge this.